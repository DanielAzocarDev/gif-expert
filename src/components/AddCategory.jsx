import { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputValue = (e) => {
    setInputValue(e.target.value);
  };

  const handlerSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      //utilizando un callback puedo hacer referencia al estado anterior del setCategories sin pasale la propiedad del estado actual al component
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  };
  return (
    <form onSubmit={handlerSubmit}>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputValue}
        name=""
        id=""
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
