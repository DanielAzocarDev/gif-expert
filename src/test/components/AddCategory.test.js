import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import AddCategory from "../../components/AddCategory"


describe('test <AddCategory/>', () => { 
  const setCategories = jest.fn()
  let wrapper = shallow(<AddCategory setCategories={setCategories} />)

  beforeEach(()=> {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })
  test('should render the component', () => { 
    expect(wrapper).toMatchSnapshot()
  })
  test('should change input', () => { 
    const input = wrapper.find('input')
    const value = 'Hi!'
    input.simulate('change', {target:{value}})
  })
  test('should not submit data', () => { 
    const form = wrapper.find('form')

    form.simulate('submit', {preventDefault(){}})

    expect(setCategories).not.toHaveBeenCalled()
  })
  test('should submit data and make sure that input value is empty', () => { 
    const input = wrapper.find('input')
    let value = 'Hi!'
    input.simulate('change', {target:{value}})

    const form = wrapper.find('form')
    form.simulate('submit', {preventDefault(){}})

    expect(setCategories).toHaveBeenCalled()

    

    expect(wrapper.find('input').prop('value')).toBe('')
  })
 })