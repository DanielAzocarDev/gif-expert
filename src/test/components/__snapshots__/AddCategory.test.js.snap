// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <AddCategory/> should render the component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    id=""
    name=""
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
